litgpt/model.py

```diff
@@ -129,8 +129,9 @@ def rope_cache(self, device: Optional[torch.device] = None) -> Tuple[torch.Tenso
                 }
             else:
                 # Some but not all parameters are specified; raise an error
+                missing_params = [param for param, present in zip(adjusted_params_required, params_present) if not present]
```

Comment:
I thought that zip in case of iterators of unequal length will iterate till the shortest one.
Maybe you wanted to use itertools.zip_longest, but then how can you guarantee that items in the list will match.

Perhaps convert to sets and return difference.

Reply:
Or you can rewrite the code a bit.
Iterate over required keys and if it is in the config - add it to a dict. If not - to the list of missing keys.

Reply:
So weird, I tested it yesterday and could swear it worked. But looking at it again with fresh eyes, I do agree with your point about `zip`.

Reply:
Again 🙂 
Yes, you are right. It's a list of booleans and not a list of keys, so the length should be the same.
It's just more of a mask, rather than a list of params that are presented in the config. That what confused me.